src/Api/Monetization/Structure/Reports/Criteria/AbstractCriteria.php

```diff
@@ -125,7 +119,10 @@ public function getApps(): array
     /**
      * @param string ...$appIds
      *
-     * @return \self
+     * @return self
+     *
+     * @deprecated in 3.0.7, will be removed in 4.0.0. No longer needed.
```

Comment:
Don't we also have a habit to  trigger runtime deprecation via trigger_error()?

Reply:
Yes we do.
Okay let me update the PR.